docs/how-does-version-resolution-work-in-maven-and-gradle.md

```diff
@@ -0,0 +1,108 @@
+# How does version resolution work in Maven and Gradle?
+
+The build system that you choose for building your Java code determines which
+version selection algorithm is used to choose the versions of your
```

Comment:
usually called the dependency mediation algorithm, at least in Maven

Reply:
In Gradle it is called "version conflict resolution": https://docs.gradle.org/current/userguide/dependency_resolution.html

I will make a note of the two terms.